### PR TITLE
Refactor stats bar to be more responsive on small-medium screen sizes

### PR DESCRIPTION
**Pull request by maurizi**
-  Fixes responsive styling issue with gray bar on screens smaller than 990px
 -  Combines add tree and add resource buttons into a single button group

screenshot

With these changes, it should not be possible for items to overlap or get pushed down in the header, no matter the screen size or the length of the "resources" terminology.

Connects to #2737

### CODE REVIEW

File: opentreemap/treemap/js/src/lib/addResourceMode.js [hunk 1]
```diff
@@ -120,11 +120,15 @@ function init(options) {
         U.$find('select', $form).on('change', enableFinalStep);
     }
```
Comment (RobinIsTheBird) [inline opentreemap/treemap/js/src/lib/addResourceMode.js]: Not sure if this is significant, but **Add a Bioswale** causes lines 118 and 120 to generate warnings in `utility.js` that their respective selectors are not found.

File: opentreemap/treemap/templates/treemap/partials/add_plot_btn.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% load i18n %}
```
Comment (maurizi) [inline opentreemap/treemap/templates/treemap/partials/add_plot_btn.html]: I think the name right now is accurate enough - adding a tree necessarily entails adding a plot as well, and in fact the add process won't create a tree if you do not fill in tree fields.

Also, I don't think changing the name is enough of an improvement to refactor the code that is relying on the current value of the `data-feature` attribute.

File: opentreemap/treemap/templates/instance_base.html [hunk 3]
```diff
@@ -134,8 +120,8 @@
   {% endif %}
 
   <div class="stats-bar hidden-xs">
-    <div class="half">
-      <div style="display: inline;" id="tree-and-planting-site-counts">
+    <div class="stats-list">
+      <div id="tree-and-planting-site-counts">
```
Comment (maurizi) [inline opentreemap/treemap/templates/instance_base.html]: That is more accurate, but there is JS code that uses this `id`, and I don't think it's a great use of our time to go changing it everywhere it is used.

File: opentreemap/treemap/js/src/lib/urlState.js [hunk 4]
```diff
@@ -185,6 +195,13 @@ module.exports = {
         });
```
Comment (maurizi) [inline opentreemap/treemap/js/src/lib/urlState.js]: Looks like there is already an open issue for this https://github.com/OpenTreeMap/otm-core/issues/2784

File: assets/css/sass/partials/_subheader.scss [hunk 5]
```diff
@@ -187,31 +187,32 @@
...
-                float: right;
+            .exportBtn {
+                margin-left: 10px;
+                margin-top: -12px;
+                width: 135px;
+                display: inline-block;
+            }
+            #tree-and-planting-site-counts {
+                max-width: calc(100% - 305px);
+                display: inline-block;
```
Comment (maurizi) [inline assets/css/sass/partials/_subheader.scss]: Honestly I arrived at most of these pixel amounts via trial-and-error, not from adding up the widths and margins of various other things. I added a comment to this effect in https://github.com/OpenTreeMap/otm-core/pull/2787/commits/e97505c17612282f7555c11a6cf045703589080e

File: opentreemap/treemap/js/src/lib/urlState.js [hunk 6]
```diff
@@ -80,6 +80,12 @@ var serializers = {
         if (state.modeName) {
             query.m = state.modeName;
         }
+    },
+
+    modeType: function(state, query) {
+        if (state.modeType) {
+            query.t = state.modeType;
+        }
```
Comment (maurizi) [inline opentreemap/treemap/js/src/lib/urlState.js]: Fixed in https://github.com/OpenTreeMap/otm-core/pull/2776

**Comment by RobinIsTheBird** [comment]
:+1:

**Pull request merged**